src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja

```diff
@@ -12,107 +14,144 @@ except ModuleNotFoundError as exc:
+                if extension_fn.object_name.startswith(snowpark_utils.TEMP_OBJECT_NAME_PREFIX):
+                    # The object name is a generated one, don't use it
+                    return None
+
+        except Exception:
+            # ignore any exception and fall back to using the object name reported from Snowpark
+            pass
+
+        return extension_fn.object_name
+
+    def __snowflake_internal_create_package_list(extension_fn):
+        return [pkg_spec.strip() for pkg_spec in extension_fn.all_packages.split(",")]
+
+    def __snowflake_internal_make_extension_fn_signature(extension_fn):
+        # Try to fetch the original argument names from the extension function
+        try:
+            args_spec = inspect.getfullargspec(extension_fn.func)
+            original_arg_names = args_spec[0]
+            start_index = len(original_arg_names) - len(extension_fn.input_sql_types)
+            signature = []
+            defaults_start_index = len(original_arg_names) - len(args_spec.defaults or [])
+            for i in range(len(extension_fn.input_sql_types)):
+                arg = {
+                    'name': original_arg_names[start_index + i],
+                    'type': extension_fn.input_sql_types[i]
+                }
+                if i >= defaults_start_index:
+                    arg['default'] = args_spec.defaults[defaults_start_index + i]
+                signature.append(arg)
+
+            return signature
+        except Exception as e:
+            msg = str(e)
+            pass # ignore, we'll use the fallback strategy
+
+        # Failed to extract the original arguments through reflection, fall back to alternative approach
+        return [
+            {"name": input_arg.name, "type": input_type}
+            for (input_arg, input_type) in zip(extension_fn.input_args, extension_fn.input_sql_types)
+        ]
+
+    def __snowflake_internal_extension_fn_to_json(extension_fn):
+        if not isinstance(extension_fn.func, Callable):
+            # Unsupported case: extension function is a tuple
+            return
+
+        # Collect basic properties of the extension function
+        extension_fn_json = {
+            "function_type": extension_fn.object_type.name,
+            "lineno": __snowflake_internal_try_extract_lineno(extension_fn.func),
+            "name": __snowflake_internal_extract_name(extension_fn),
+            "handler": extension_fn.func.__name__,
+            "imports": extension_fn.all_imports or [],
+            "packages": __snowflake_internal_create_package_list(extension_fn),
+            "runtime": extension_fn.runtime_version,
+            "returns": extension_fn.return_sql.upper().replace("RETURNS ", "").strip(),
+            "signature": __snowflake_internal_make_extension_fn_signature(extension_fn),
+            "external_access_integrations": extension_fn.external_access_integrations or [],
+            "secrets": extension_fn.secrets or {},
+        }
```

Comment:
Some missing keys:
1. raw_imports: unvalidated here is fine, the validation is part of the weaving DDL PR. We do need access to the raw imports that a user has passed into the decorator, because it may or may not be the path to this exact file. And in our case, with Session being None always, `all_imports` is guaranteed to be empty.
2. anonymous: we need to check this, either here in this file or later in generation, about anonymous procedures because we skip generating DDL for that.

Reply:
I left raw imports out on purpose. We can discuss if you want. Anonymous should be filtered out here instead of exposing it through collected functions.

Reply:
+1, let's have this discussion offline.